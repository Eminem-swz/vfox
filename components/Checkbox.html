<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Checkbox/CheckboxGroup 多选项 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.e3398b0e.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <link rel="modulepreload" href="/vfox/assets/components_Checkbox.md.91facf5f.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.9b8f0faa.js">
    <meta name="twitter:title" content="Checkbox/CheckboxGroup 多选项 | Vfox for Vue3">
    <meta property="og:title" content="Checkbox/CheckboxGroup 多选项 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Checkbox">Checkbox</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkbox-props">Checkbox Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkbox-events">Checkbox Events</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkbox-slots">Checkbox Slots</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkboxgroup">CheckboxGroup</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkboxgroup-props">CheckboxGroup Props</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#option">Option</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkboxgroup-events">CheckboxGroup Events</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#checkboxgroup-slots">CheckboxGroup Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）">默认（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Checkbox" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="checkbox-checkboxgroup-多选项" tabindex="-1">Checkbox/CheckboxGroup 多选项 <a class="header-anchor" href="#checkbox-checkboxgroup-多选项" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-"><pre><code>import { Checkbox, CheckboxGroup } from &#39;vfox&#39;
</code></pre></div><p>具体的引入方式可以参考<a href="./../#引入组件">引入组件</a>。</p><h2 id="checkbox-props" tabindex="-1">Checkbox Props <a class="header-anchor" href="#checkbox-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>value</td><td>string/number</td><td>&#39;&#39;</td><td>否</td><td>该项值，<a href="./Checkbox.html#checkboxgroup-多项选择器">CheckboxGroup</a> 的 change 事件会携带 <code>checkbox</code> 的 value</td></tr><tr><td>disabled</td><td>boolean</td><td>false</td><td>否</td><td>是否禁用</td></tr><tr><td>v-model:checked</td><td>boolean</td><td>false</td><td>否</td><td>当前是否选中，可用来设置默认选中</td></tr><tr><td>activeColor</td><td>string</td><td></td><td>否</td><td>自定义激活态的图标颜色</td></tr></tbody></table><h2 id="checkbox-events" tabindex="-1">Checkbox Events <a class="header-anchor" href="#checkbox-events" aria-hidden="true">#</a></h2><table><thead><tr><th>事件名称</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>change</td><td>勾选发生改变时触发</td><td>{ checked: boolean }</td></tr></tbody></table><h2 id="checkbox-slots" tabindex="-1">Checkbox Slots <a class="header-anchor" href="#checkbox-slots" aria-hidden="true">#</a></h2><div class="language-"><pre><code>&lt;fx-checkbox&gt;开朗&lt;/fx-checkbox&gt;
&lt;fx-checkbox&gt;活泼&lt;/fx-checkbox&gt;
...
</code></pre></div><h2 id="checkboxgroup" tabindex="-1">CheckboxGroup <a class="header-anchor" href="#checkboxgroup" aria-hidden="true">#</a></h2><p>注：</p><ul><li>内部由多个 <a href="./Checkbox.html#Checkbox-多选项">Checkbox</a> 组成。</li><li>支持表单，具体可参考 <a href="./Form.html">Form</a>。</li></ul><h2 id="checkboxgroup-props" tabindex="-1">CheckboxGroup Props <a class="header-anchor" href="#checkboxgroup-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>name</td><td>string</td><td></td><td>否</td><td>标识</td></tr><tr><td>inline</td><td>boolean</td><td>false</td><td>否</td><td>是否使用内联布局，默认纵向布局</td></tr><tr><td>v-model</td><td>string[]/number[]</td><td>[]</td><td>否</td><td>当前选择子项的 value 列表</td></tr><tr><td>activeColor</td><td>string</td><td></td><td>否</td><td>自定义子项激活态的图标颜色</td></tr><tr><td>options</td><td>string[]/number[]/<a href="./Checkbox.html#option">Option</a>[]</td><td></td><td>否</td><td>子项配置，优先级低于 slot 放入 <code>Checkbox</code> 组件</td></tr></tbody></table><h3 id="option" tabindex="-1">Option <a class="header-anchor" href="#option" aria-hidden="true">#</a></h3><table><thead><tr><th>Option Key</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>value</td><td>string/string</td><td></td><td>是</td><td>值， 同 <code>Checkbox</code> 组件 value</td></tr><tr><td>label</td><td>string</td><td></td><td>是</td><td>描述，同 <code>Checkbox</code> 组件 slot default</td></tr></tbody></table><h2 id="checkboxgroup-events" tabindex="-1">CheckboxGroup Events <a class="header-anchor" href="#checkboxgroup-events" aria-hidden="true">#</a></h2><table><thead><tr><th>事件名称</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>change</td><td>选中项发生改变时触发</td><td>value: (string | number)[]</td></tr></tbody></table><h2 id="checkboxgroup-slots" tabindex="-1">CheckboxGroup Slots <a class="header-anchor" href="#checkboxgroup-slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-checkbox-group&gt;
  &lt;fx-checkbox&gt;开朗&lt;/fx-checkbox&gt;
  &lt;fx-checkbox&gt;活泼&lt;/fx-checkbox&gt;
  ...
&lt;/fx-checkbox-group&gt;
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/Radio" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Radio</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/Slider" data-v-38ede35f><span class="text" data-v-38ede35f>Slider</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"2625163a\",\"apis_modal.md\":\"be15069e\",\"apis_notify.md\":\"882f3e00\",\"apis_popover.md\":\"481e2777\",\"apis_preview.md\":\"d4e45991\",\"apis_toast.md\":\"611b8337\",\"apis_index.md\":\"28b5f737\",\"components_actionsheet.md\":\"2d3adb3f\",\"components_activityindicator.md\":\"5ecea181\",\"components_avatar.md\":\"2ae79c5f\",\"components_backtop.md\":\"f9caa6ad\",\"components_badge.md\":\"4d4bfa75\",\"components_button.md\":\"75c70f34\",\"components_calendar.md\":\"cff1f671\",\"components_cascader.md\":\"3672a71e\",\"components_cell.md\":\"cde5b293\",\"components_checkbox.md\":\"91facf5f\",\"components_circleprogress.md\":\"a64da26d\",\"components_collapse.md\":\"24edf03e\",\"components_copy.md\":\"25a4b205\",\"components_countdown.md\":\"835dacb6\",\"components_countup.md\":\"820c3042\",\"components_datepicker.md\":\"cde17a48\",\"components_dialog.md\":\"ff4e0619\",\"components_divider.md\":\"18efc9c0\",\"components_drawer.md\":\"db007bca\",\"components_dropdown.md\":\"042b00f2\",\"components_empty.md\":\"213d88cb\",\"components_fixed.md\":\"9d209e4b\",\"components_flatlist.md\":\"36fcaac4\",\"components_form.md\":\"3e82d8cb\",\"components_group.md\":\"07998a1a\",\"components_icon.md\":\"8c3caebd\",\"components_image.md\":\"db3c7480\",\"components_imagepreview.md\":\"ac196467\",\"components_imageuploader.md\":\"2dbbef4b\",\"components_indexview.md\":\"1ac7c899\",\"components_input.md\":\"51d38b41\",\"components_layout.md\":\"db407131\",\"components_loadmore.md\":\"56cbcc9c\",\"components_modal.md\":\"69b30342\",\"components_navbar.md\":\"127293f3\",\"components_noticebar.md\":\"09f9d5b9\",\"components_notify.md\":\"9d238fdd\",\"components_numberkeyboard.md\":\"c9870a3e\",\"components_order.md\":\"1a920353\",\"components_pagination.md\":\"8edda886\",\"components_picker.md\":\"8e79b245\",\"components_popdialog.md\":\"6d0daecd\",\"components_popmenu.md\":\"fc15c9e7\",\"components_popover.md\":\"97e280d8\",\"components_price.md\":\"5ae042ab\",\"components_progress.md\":\"20b863da\",\"components_radio.md\":\"46196553\",\"components_range.md\":\"134699f9\",\"components_rate.md\":\"cd4bc273\",\"components_result.md\":\"02cb75a5\",\"components_scrolltab.md\":\"42e11f68\",\"components_scrollview.md\":\"c3be3a69\",\"components_searchbar.md\":\"c1e34c25\",\"components_sidetab.md\":\"863b195f\",\"components_skeleton.md\":\"4f42e805\",\"components_slider.md\":\"3b33940d\",\"components_stepper.md\":\"aa23322a\",\"components_steps.md\":\"3f40da35\",\"components_sticky.md\":\"9fa44c01\",\"components_stopwatch.md\":\"4306f52a\",\"components_swipecell.md\":\"75c23dcf\",\"components_swiper.md\":\"e708df08\",\"components_switch.md\":\"3eab06f8\",\"components_tab.md\":\"4b9d3cb0\",\"components_tabbar.md\":\"15ec38ec\",\"components_tabview.md\":\"c3b165d4\",\"components_tag.md\":\"b7083284\",\"components_timeago.md\":\"ed332239\",\"components_timeline.md\":\"529ea4f5\",\"components_toast.md\":\"aeaaec7d\",\"components_index.md\":\"7489b92b\",\"design_color.md\":\"2df0d812\",\"design_index.md\":\"1ad33247\",\"index.md\":\"100f2b2a\"}")</script>
    <script type="module" async src="/vfox/assets/app.9b8f0faa.js"></script>
    
  </body>
</html>