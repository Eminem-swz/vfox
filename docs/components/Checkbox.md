# Checkbox 多选项

## Checkbox Props

| 属性            | 类型          | 默认值 | 必填 | 说明                                                                                                      |
| --------------- | ------------- | ------ | ---- | --------------------------------------------------------------------------------------------------------- |
| value           | string/number | ''     | 否   | 该项值，[checkbox-group](./Checkbox.md#checkboxgroup-多项选择器) 的 change 事件会携带 `checkbox` 的 value |
| disabled        | boolean       | false  | 否   | 是否禁用                                                                                                  |
| v-model:checked | boolean       | false  | 否   | 当前是否选中，可用来设置默认选中                                                                          |

## Checkbox CSS

| 属性                    | 默认值  | 说明       |
| ----------------------- | ------- | ---------- |
| --checkbox-color        | #d9d9d9 | 图标色     |
| --checkbox-active-color | #1890ff | 图标激活色 |
| --checkbox-font-color   | #262626 | 文本色     |

## Checkbox Events

| 事件名称 | 描述               | 回调函数参数         |
| -------- | ------------------ | -------------------- |
| change   | 勾选发生改变时触发 | { checked: boolean } |

## Checkbox Slots

```
<fx-checkbox>开朗</fx-checkbox>
<fx-checkbox>活泼</fx-checkbox>
...
```

## CheckboxGroup 多项选择器

- 内部由多个 [Checkbox](./Checkbox.md#Checkbox-多选项) 组成。

- 可以配合 [Form](./Form.md) 和 [FormItem](./Form.md#formitem-表单项) 使用。

## CheckboxGroup Props

| 属性    | 类型              | 默认值 | 必填 | 说明                           |
| ------- | ----------------- | ------ | ---- | ------------------------------ |
| name    | string            |        | 否   | [Form](./Form.md) 的标识       |
| inline  | boolean           | false  | 否   | 是否使用内联布局，默认纵向布局 |
| v-model | string[]/number[] | []     | 否   | 当前选择子项的 value 列表      |

## CheckboxGroup Events

| 事件名称 | 描述                 | 回调函数参数 |
| -------- | -------------------- | ------------ |
| change   | 选中项发生改变时触发 | { value }    |

## CheckboxGroup Slots

### 默认（#default）

```
<fx-checkbox-group>
  <fx-checkbox>开朗</fx-checkbox>
  <fx-checkbox>活泼</fx-checkbox>
  ...
</fx-checkbox-group>
```
