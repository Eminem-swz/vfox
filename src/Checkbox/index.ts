import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Checkbox from './Checkbox.vue'

const _Checkbox: SFCWithInstall<typeof Checkbox> = Object.assign(Checkbox, {
  install: function (app: App) {
    app.component(Checkbox.name, Checkbox)
  }
})

export default _Checkbox
