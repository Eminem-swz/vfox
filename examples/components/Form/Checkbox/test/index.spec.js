/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Checkbox from '@/components/Checkbox'

describe('Checkbox', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Checkbox, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
