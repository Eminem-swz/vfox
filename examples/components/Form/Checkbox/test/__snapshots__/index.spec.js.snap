// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render default correctly 1`] = `
<label class="fx-checkbox fx-horizontal-hairline"><input class="fx-checkbox_input" type="checkbox" name="" value="">
  <div class="fx-checkbox_box"><svg class="fx-icon fx-checkbox_icon">
      <use xlink:href="#icon-BorderOutlined"></use>
    </svg><svg class="fx-icon fx-checkbox_checked-icon">
      <use xlink:href="#icon-CheckSquareFilled"></use>
    </svg>
    <!--v-if-->
  </div>
</label>
`;
